crates/ide/src/syntax_highlighting/tests.rs

```diff
@@ -1435,3 +1525,173 @@ fn main() {
         false,
     );
 }
+
+#[test]
+fn test_comment_highlighting_disabled() {
+    // Test that comments are not highlighted when disabled
+    check_highlighting_with_config(
+        r#"
+// This is a regular comment
+/// This is a doc comment
+fn main() {
+    // Another comment
+    println!("Hello, world!");
+}
+"#,
+        HighlightConfig {
+            strings: true,
+            comments: false,     // Disable comment highlighting
+            doc_comments: false, // Also disable doc comment highlighting
+            punctuation: true,
+            specialize_punctuation: true,
+            specialize_operator: true,
+            operator: true,
+            inject_doc_comment: true,
+            macro_bang: true,
+            syntactic_name_ref_highlighting: false,
+        },
+        expect_file!["./test_data/highlight_comments_disabled.html"],
+        false,
+    );
+}
+
+#[test]
+fn test_comment_highlighting_enabled() {
```

Comment:
We don't need to test the enabled case, other tests cover that.

Reply:
Removed it.